Add Functions To Canonicalize Prometheus Metric Names

pamaddox opened the pull request:
Added two new functions, `noit_prometheus_translate_snappy_data_canonical` and `noit_prometheus_metric_name_from_labels_canonical`, that will guarantee that the caller gets fully canonical metric names.

pamaddox committed:
Always Canonicalize Prometheus Metric Names

There was a comment that there was no need to canonicalize metric names
here because the system would handle it automatically; however, this is
a library function, which means we do not know how the caller will use
the data. We need to canonicalize the metric name here so we know that
the caller gets the correct name.

postwait reviewed (changes requested):
I believe this was intentional because callsites were already canonicalizing and given the cost we didn't want to force doing it twice.

I would add another function that promises to canonicalize... `_canonical`

pamaddox committed:
Revert "Always Canonicalize Prometheus Metric Names"

This reverts commit acb2f95ea0ee36da061c4a6e313db7dad61c82ac.

pamaddox committed:
Add noit_prometheus_metric_name_from_labels_canonical

Add new function, noit_prometheus_metric_name_from_labels_canonical,
which will canonicalize the metric name.

pamaddox committed:
Add snappy canonical translation function
